File: programers_Lv2/다음 큰 숫자.py
def solution(n):
    answer = 0
    n_cnt = bin(n).count('1') # bin(n)은 문자열을 반환

    while True:
        n += 1
        temp_cnt = bin(n).count('1')
        if temp_cnt == n_cnt:
            answer = n
            break
    return answer

print(solution(15))

File: programers_Lv2/다음 큰 숫자.js
function solution(n) {
  var answer = 0;
  nCnt = [...n.toString(2)].filter((el) => el === "1").length;

  while (true) {
    n++;
    tempCnt = [...n.toString(2)].filter((el) => el === "1").length;
    if (tempCnt === nCnt) {
      answer = n;
      break;
    }
  }

  return answer;
}

console.log(solution(15));
